### PR TITLE
[Lock] add type-hint whenever possible

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? |no 
| Tests pass?   | yes    
| Fixed tickets | Contribute to #32179 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Add type-hint in the lock component

### CODE REVIEW

**Comment by OskarStark** [comment]
The fixed ticket should not include “t”

File: src/Symfony/Component/Lock/Strategy/StrategyInterface.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ interface StrategyInterface
      *
      * @return bool
      */
-    public function isMet($numberOfSuccess, $numberOfItems);
+    public function isMet(int $numberOfSuccess, int $numberOfItems);
```
Comment (derrabus) [inline src/Symfony/Component/Lock/Strategy/StrategyInterface.php]: You can remove the corresponding `@param` annotations.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Review by jderusse — COMMENTED** [review]
Could you please add `acquire(bool $blocking = false)` in `LockInterface`?

**Review by nicolas-grekas — APPROVED** [review]
(I'm taking care of the comments while merging)

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Comment by Simperfit** [comment]
thanks @nicolas-grekas for taking care of the comments!

**Pull request closed without merging**